Mission script

TaeSeanDo opened the pull request.

TaeSeanDo committed:
Copy/pasted from 2017 without requested exceptions.

TaeSeanDo committed:
Fixes exceptions in Mission script, getMission method in Mission.java

docprofsky reviewed (changes requested):
This all looks good, but you still need to do the rest of #2.

docprofsky committed:
Merge branch 'master' into MissionScript

docprofsky committed:
Rename getMission() to getCommands() for maximum accuracy

docprofsky committed:
Add potential implementation for a MissionScriptMission

docprofsky commented:
I added my work on the MissionScriptMission. This may give you a better idea of what my thoughts on implementing it are.

Regarding the handling of the parser output (see my comments in MissionScriptMission.java for context), it might also be useful to have the MissionScript parser return a MissionScriptMission, instead of a normal Mission.

TaeSeanDo committed:
MissionScriptMission set up.

Signed-off-by: Sean <seantierney42@gmail.com>

TaeSeanDo committed:
Implements MissionScriptMission, sets up MissionSendable, missionChooser, and DoNothing auto mission.
Still needs to use missionSendable in TeleOp.
Closes #2

TaeSeanDo committed:
Mission chooser, implemented but slightly broken. Runs Do Nothing Mission no matter what. Also whitespace.

TaeSeanDo committed:
Run selected auto mission in autonomous.

docprofsky commented:
This is also supposed to address both #2 and #3, but the chooser stuff for #3 is still not working.

docprofsky commented:
@Ben1492 Can you look this review the code to see if there is anything we may be missing?

TaeSeanDo committed:
Add MissionSendable implementation, missionChooser.

docprofsky committed:
Avoid naming issue with SendableChooser and add readback for selected mission

TaeSeanDo committed:
Correct File path.

docprofsky committed:
Fix bug with parsing whitespace in MissionScript parser
This is a problem for “moveStraight(0.5, 1, false)”.
It treats the last parameter as “ false”, which does not match “false”. Notice the extra whitespace.

docprofsky committed:
Add output for when parser returns null for a mission

docprofsky commented:
All of this has been tested and works.
@Ben1492 can you review this?

Example mission script file:
```
turnInPlace(0.3, 720)
moveStraight(0.5, 3, false)
delay(0.5)
turnInPlace(0.3, 170)
delay(1.5)
moveStraight(0.5, 2, false)
```

docprofsky committed:
Clean up whitespace

docprofsky committed:
Add JavaDoc for MissionScriptMission
